### DISCUSSION TITLE
HKStreamRecorder: if I start recording RTMPS stream and my app crash can I fix the video file?

### DISCUSSION THREAD

**Discussion by al3x88** [HaishinKit/HaishinKit.swift#1626] [Q&A]
I noticed that if a crash occurs within my app while I am streaming and recording, the recorder file is somehow present in my tmp directory. Of course the file is not readable because the "stopRecording()" cannot be called due to the crash....

Since the file is present is there a way to fix this file making it readable? Would be very useful in case of crash!!!

**Comment by al3x88** [comment]
I was reading that when setMovieFragmentInterval is enabled (and that is the case in HaishinKit) the file should be recoverable.
As I said I see the file but I am unable to play it either with IOS Player or VLC.

I configured this way in viewdidload:

`Task {
            
            await recorder.setMovieFragmentInterval(movieFragmentInterval: 10) 

Any other settings are needed? many thanks!

**Reply by shogo4405** [reply]
I don't have any insights on this. I have confirmed that when setMovieFragmentInterval is enabled, playback works if the end user kills the task. However, I've also heard that it doesn't work well in the case of a crash. Since this uses iOS APIs, there's nothing I can do on my end.

**Reply by al3x88** [reply]
thanks for your reply.
Seems a peace of code is missing in the last version: 

`if let movieFragmentInterval = self.movieFragmentInterval { self.writer?.movieFragmentInterval = CMTime(seconds: movieFragmentInterval, preferredTimescale: 1) }`

As mentioned here: 
https://github.com/shogo4405/HaishinKit.swift/discussions/1626#discussioncomment-11291241

So basically the setting is never called from the AssetWriter...

**Comment by al3x88** [comment]
Hi the issue is that on the last version:
https://github.com/shogo4405/HaishinKit.swift/blob/main/HaishinKit/Sources/HKStream/HKStreamRecorder.swift

is missing this code:

`if let movieFragmentInterval = self.movieFragmentInterval {
                    self.writer?.movieFragmentInterval = CMTime(seconds: movieFragmentInterval, preferredTimescale: 1)
                }`

which is published here: https://github.com/shogo4405/HaishinKit.swift/pull/1561/commits/677ace5b1fe25e02e559fa2047e97a629c3bb0fc#diff-4d240a8bb1529e880be6e26f72498946dc1ed4fc8d00eb9b412f85c6e30ecae1

so basically the setting is not applied.... am I doing something wrong?

**Reply by shogo4405** [reply]
Ah, you're right. The code is indeed missing. It might have been removed during the migration to 2.0.x. I have fixed it with this PR.
https://github.com/shogo4405/HaishinKit.swift/pull/1627

**Reply by al3x88** [reply]
@shogo4405 will you merge into main branch soon? 
Many thanks!!!

**Reply by shogo4405** [reply]
Merged.

**Reply by al3x88** [reply]
@shogo4405 thanks! Will you create a new version? Seems SPM does not pull changes keeping the previous version 2.0.0

**Reply by shogo4405** [reply]
No, please use it by specifying the commit hash. The release of version 2.0.1 is not planned for the near future.

**Comment by shogo4405** [comment]
closed as fixed f433315e
